cloud-checksum/src/cli.rs

```diff
@@ -43,6 +43,15 @@ pub struct Command {
     /// Options related to outputting data from the CLI.
     #[command(flatten)]
     pub output: Output,
+    /// Options related to how cloud credentials are obtained
+    #[arg(
+        global = true,
+        long,
+        env,
+        default_value = "default-environment",
+        alias = "source-credential-provider"
+    )]
+    pub credential_provider: CredentialProvider,
```

Comment:
Thinking that for generate/check - there is just one "credential-provider". But once we get to copy we have a source and a destination. So we alias the source to this global parameter.